aws-http-nio-client-netty/src/main/java/software/amazon/awssdk/http/nio/netty/NettyNioAsyncHttpClient.java

```diff
@@ -115,6 +118,24 @@ public void close() {
         group.shutdownGracefully();
     }
 
+    /**
+     * Depending on the EventLoopGroup used we may need to use a different socket channel.
+     */
+    @ReviewBeforeRelease("Perhaps we should make the customer provide both event loop group" +
+                         "and channel in some kind of wrapper class to avoid having to do this.")
+    private Class<? extends Channel> resolveSocketChannelClass() {
```

Comment:
This is a bit janky. I was thinking we could have another container where you have to provide the EventLoopGroup and the channel class/factory in one object. The EventLoopGroupFactory would create this container.